python/tvm/ir_builder.py

```diff
@@ -197,6 +197,16 @@ def for_range(self, begin, end, name="i", dtype="int32", for_type="serial"):
         self._seq_stack.append([])
         loop_var = _api.var(name, dtype=dtype)
         extent = end if begin == 0 else _pass.Simplify(end - begin)
+
+        if for_type == "serial":
```

Comment:
I think this implicit modification makes thing a bit opaque to developer. Also will it always be better to do such a transformation?